fdbserver/DataDistribution.actor.cpp

```diff
@@ -4486,6 +4514,45 @@ ACTOR Future<Void> ddSnapCreate(DistributorSnapRequest snapReq, Reference<AsyncV
 	return Void();
 }
 
+ACTOR Future<Void> ddExclusionSafetyCheck(DistributorExclusionSafetyCheckRequest req,
+                                          Reference<DataDistributorData> self, Database cx) {
+	TraceEvent("DDExclusionSafetyCheckBegin");
+	vector<StorageServerInterface> ssis = wait(getStorageServers(cx));
+	DistributorExclusionSafetyCheckReply reply(true);
+	if (!self->teamCollection) {
```

Comment:
Does this ever happen?
The `self->teamCollection` should always be set in the `dataDistribution(self)` function in `dataDistributor` actor. 

Can you add an ASSERT() here and check if this ever happens? 
If it's confirmed not happening, we do not need to introduce the `teamCollection` pointer to the `DataDistributorData`

Reply:
Yes, this check is necessary. I've observed this happening several times in simulation.

Reply:
Interesting. I suggest putting this into the TODO list because I want to figure out when this happens. It may indicate some bug in DD.